### PR TITLE
ref(structure): moving plugins into a separate dir

### PR DESCRIPTION
**Pull request by SBub**
Note: transport file was removed and broken down into separate plugin files. Backward compatibility is broken if someone was exporting contruct/implementation of i.e. JolocomLinking. Next release should happened with the major version update

### CODE REVIEW

File: ts/index.ts [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
+/**
+ * Re-exporting available plugins
+ */
+export { DeepLinkingProvider } from './plugins/deep-linking/DeepLinkingProvider'
+export { JolocomDeepLinkingTransport } from './plugins/deep-linking/JolocomDeepLinkingTransport'
+export {
+  LinkingErrorCode,
+  JolocomLinking,
+} from './plugins/linking/JolocomLinking'
+export { JolocomWebSockets } from './plugins/web-sockets/JolocomWebSockets'
```
Comment (clauxx) [inline ts/index.ts]: I think the additional nesting doesn't bring any value here. Would be better if all the plugins lay directly under `/plugins`.

File: ts/plugins/linking/JolocomLinking.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import {
+  InteractionTransportType,
+  JolocomPlugin,
+  JolocomSDK,
+  TransportAPI,
+  TransportDesc,
+} from '@jolocom/sdk'
+import { Linking } from 'react-native'
+
+export enum LinkingErrorCode {
```
Comment (clauxx) [inline ts/plugins/linking/JolocomLinking.ts]: Better to have a separate `ReactNativeJolocomError` (or smth like that) enum instead somewhere at the top level.

File: ts/index.ts [hunk 3]
```diff
@@ -1,27 +1,40 @@
...
 export * from './storage'
 export * from './passwordStore'
 export * from './secureStorage'
 export * from '@jolocom/sdk'
 export * from 'jolocom-lib'
+
+/**
+ * Re-exporting available plugins
+ */
+export { DeepLinkingProvider } from './plugins/deep-linking/DeepLinkingProvider'
```
Comment (clauxx) [inline ts/index.ts]: To make exporting easier, we can have an `index.ts` inside `/plugins` which aggregates and exports all the plugins. As a result, here we can just do `export * from ./plugins`.

**Review by clauxx — CHANGES_REQUESTED** [review]

File: ts/plugins/linking/JolocomLinking.ts (hunk 2 above)
Comment (SBub) [reply ts/plugins/linking/JolocomLinking.ts]: It is available at the top level, as it is reexported from the index

File: ts/index.ts (hunk 3 above)
Comment (SBub) [reply ts/index.ts]: I won't to avoid using *, it become really un-readable was is being reexported

**Review by clauxx — APPROVED** [review]

**Pull request merged**